contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -5325,3 +5336,169 @@ pltsql_remove_current_query_env(void)
 		destroy_failed_transactions_map();
 	} 
 }
+
+/*
+ * Drop statement of babelfish, currently delete extended property as well.
+ */
+static void
+bbf_ExecDropStmt(DropStmt *stmt)
+{
+	int16			db_id;
+	char			*physical_schema_name = NULL,
+					*schema_name = NULL,
+					*major_name = NULL;
+	const char		*logical_schema_name = NULL;
+	ObjectAddress	address;
+	Relation		relation = NULL;
+	Oid				schema_oid;
+	ListCell		*cell;
+
+	db_id = get_cur_db_id();
+
+	if (stmt->removeType == OBJECT_SCHEMA)
+	{
+		foreach(cell, stmt->objects)
+		{
+			/* Get schema_name */
+			logical_schema_name = get_logical_schema_name(strVal(lfirst(cell)),
+														  true);
+
+			if (logical_schema_name)
+			{
+				schema_name = downcase_identifier(logical_schema_name,
+												  strlen(logical_schema_name),
+												  false, false);
+				pfree((char *) logical_schema_name);
+			}
+
+			if (schema_name)
+			{
+				delete_extended_property(2, db_id, schema_name, NULL, NULL, NULL);
+				pfree(schema_name);
+			}
+		}
+	}
+	else if (stmt->removeType == OBJECT_TABLE ||
+			 stmt->removeType == OBJECT_VIEW ||
+			 stmt->removeType == OBJECT_SEQUENCE)
+	{
+		foreach(cell, stmt->objects)
+		{
+			address = get_object_address(stmt->removeType,
+										 lfirst(cell),
+										 &relation,
+										 AccessShareLock,
+										 true);
+
+			if (!relation)
+				return;
+
+			/* Get major_name */
+			major_name = pstrdup(RelationGetRelationName(relation));
```

Comment:
here we may still have the problem of original name issue.  
And I feel that we don't really need to get schema and object name through Relation. 
They could be obtained just based on the object name and current schema name if needed.

Reply:
I can pass wrong schema name and wrong object name, which will cause useless search on the extended property catalog table.
It doesn't cost much through relation, cause the relation cache will be reused in PG utility function.